### PR TITLE
fix: Navigation Bar Translucency in Multiple View Controllers - WPB-11794

### PR DESCRIPTION
**Pull request by agisilaos**
This PR addresses an issue where the UINavigationController‘s navigationBar was translucent by default, leading to unexpected behavior in the navigation bar’s appearance across several view controllers. Specifically:

- Affected View Controllers: ConversationListNavigationController, ArchiveNavigationController, and SettingsNavigationController
- **Issue**: When the `isTranslucent` property is left as true, the color of the navigation bar can become transparent, causing it to take on the color of whatever content is underneath. This leads to incorrect visual behavior, particularly in views like the SettingsNavigationController and ArchiveNavigationController, where the navigation bar color was incorrect even before scrolling.

**Fix**:
- Set navigationBar.isTranslucent = false for each of the affected UINavigationControllers to ensure the navigation bar has a solid color that does not change based on the underlying content.

**Impact**:

- This change guarantees that the navigation bar maintains the correct appearance in all contexts, resolving any visual inconsistencies. Without this fix, the navigation bar in SettingsNavigationController and ArchiveNavigationController was incorrectly colored even without any interaction or scrolling.

## BEFORE

https://github.com/user-attachments/assets/2a05d5ad-0d42-4de7-974f-4bc3da16cd32

## AFTER

https://github.com/user-attachments/assets/a7a0634c-d97a-414e-bf21-c55881dff97b

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

**Review by caldrian — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift [hunk 1]
```diff
@@ -84,6 +85,14 @@ final class ConversationListViewControllerTests: XCTestCase {
             selfProfileViewControllerBuilder: .mock
         )
         sut.mainSplitViewState = .collapsed
+        // sut.navigationController?.navigationBar.barTintColor = SemanticColors.View.backgroundDefaultWhite
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift]: To delete?

File: wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift [hunk 2]
```diff
@@ -84,6 +85,14 @@ final class ConversationListViewControllerTests: XCTestCase {
...
+
+        // Set the navigation bar color here
+           let navBarAppearance = UINavigationBarAppearance()
+           navBarAppearance.backgroundColor = SemanticColors.View.backgroundDefaultWhite
+           UINavigationBar.appearance().standardAppearance = navBarAppearance
+           UINavigationBar.appearance().scrollEdgeAppearance = navBarAppearance
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift]: Won't this set all nav bar appearances? Won't this affect other tests?

**Comment by samwyndham** [comment]
Removed myself as @caldrian and @johnxnguyen are are already reviewing

File: wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift]: Done: https://github.com/wireapp/wire-ios/commit/2c669b9a98137eb0ce0aa305e9e3d9731d5bed5d

File: wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerSnapshotTests.swift]: Done: https://github.com/wireapp/wire-ios/commit/2c669b9a98137eb0ce0aa305e9e3d9731d5bed5d

Deleted all the setup styling code for now. Still trying to figure out why the generated ones come with a transparent nav bar background color.

File: WireUI/Tests/WireMainNavigationUITests/Resources/ReferenceImages/MainSplitViewControllerTests/testCompactAppearance.dark.png
Comment (johnxnguyen) [inline WireUI/Tests/WireMainNavigationUITests/Resources/ReferenceImages/MainSplitViewControllerTests/testCompactAppearance.dark.png:1]: I can't tell if these snapshots are correct

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController.swift [hunk 3]
```diff
@@ -376,7 +387,7 @@ final class ConversationListViewController: UIViewController {
     func applyColorTheme() {
         view.backgroundColor = mainSplitViewState == .expanded
         ? ColorTheme.Backgrounds.backgroundVariant
-        : ColorTheme.Backgrounds.surfaceVariant
+        : ColorTheme.Backgrounds.surface
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController.swift:390]: Does this change match the Figma design?

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController.swift:390]: Yes!

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController.swift:390]: Coming from Wolfgang:

> We should to make use of the semantic color variable “Surface” (white on light mode, gray-95 on dark mode) if possible. For background color.

File: WireUI/Tests/WireMainNavigationUITests/Resources/ReferenceImages/MainSplitViewControllerTests/testCompactAppearance.dark.png
Comment (agisilaos) [reply WireUI/Tests/WireMainNavigationUITests/Resources/ReferenceImages/MainSplitViewControllerTests/testCompactAppearance.dark.png:1]: They're we need to find a better way of testing those.

**Pull request merged**